### PR TITLE
privacyIDEA: 3.5.2 -> 3.6

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

New release: https://github.com/privacyidea/privacyidea/releases/tag/v3.6

###### Details

*<del> c022e2d (python3Packages.sqlsoup: fix build)
  This doesn't work with sqlalchemy 1.4. Yes, this project seems mostly
  dead, but it's still needed by `privacyidea`. I already reported the
  problem to upstream[1].

  [1] privacyidea/privacyidea#2790</del>

* e220c59 (privacyIDEA: 3.5.2 -> 3.6)
  ChangeLog: https://github.com/privacyidea/privacyidea/releases/tag/v3.6

  Unfortunately we have to use `sqlalchemy` at 1.3 for `sqlsoup`. As
  `sqlalchemy` is required by a lot of packages, I decided to move this
  package out of `pythonPackages` itself and instantiate a new
  `pythonPackages` inside the expression where `sqlalchemy` points to
  `sqlalchemy_1_3`.

-----

Also deployed this to my self-hosted privacyIDEA without any problems.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/development/python-modules/privacyidea/default.nix [hunk 1]
```diff
@@ -1,43 +1,37 @@
...
     rev = "v${version}";
-    sha256 = "sha256-k2om2LjkFRCT53ECPAJEztCiMdz4fF5eoipVUvSoyGo=";
+    sha256 = "sha256-yywkQ3TdBzRMbJGY0Seaprztgt0JrCAbgqosMQ5fcQM=";
     fetchSubmodules = true;
   };
 
-  patches = [
-    # Subset of https://github.com/privacyidea/privacyidea/commit/359db6dd10212b8a210e0a83536e92e9e796a1f8,
-    # fixes app context errors in tests. Can be removed on the next bump.
-    ./fix-tests.patch
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/privacyidea/default.nix:27]: Please delete this file.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/privacyidea/default.nix [hunk 2]
```diff
@@ -61,7 +55,7 @@ buildPythonPackage rec {
   '';
 
   postInstall = ''
-    rm -rf $out/${python.sitePackages}/tests
+    rm -rf $out/${python3'.sitePackages}/tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/privacyidea/default.nix:58]:
```suggestion
    rm -r $out/${python3'.sitePackages}/tests
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -13875,6 +13875,8 @@ in
 
   premake = premake4;
 
+  privacyidea = callPackage ../development/python-modules/privacyidea { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:13878]: Can you move the files under application somewhere?

**Review by lheckemann — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Pull request merged**